sn_registers/src/address.rs

```diff
@@ -6,42 +6,95 @@
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
+use crate::error::{Error, Result};
+
+use bls::{PublicKey, PK_SIZE};
 use serde::{Deserialize, Serialize};
-use std::hash::Hash;
-use xor_name::XorName;
+use std::{fmt::Display, hash::Hash};
+use xor_name::{XorName, XOR_NAME_LEN};
 
 /// Address of a Register on the SAFE Network
 #[derive(Clone, Copy, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize, Debug)]
 pub struct RegisterAddress {
-    /// Name.
-    pub name: XorName,
-    /// Tag.
-    pub tag: u64,
+    /// User chosen meta, can be anything, the register's name on the network will be the hash of this meta and the owner
+    pub(crate) meta: XorName,
+    /// Owner of the register
+    pub(crate) owner: PublicKey,
+}
+
+impl Display for RegisterAddress {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.to_hex())
+    }
 }
 
 impl RegisterAddress {
-    /// Construct a new `RegisterAddress` given `name` and `tag`.
-    pub fn new(name: XorName, tag: u64) -> Self {
-        Self { name, tag }
+    /// Construct a new `RegisterAddress` given `meta` and `owner`.
+    pub fn new(meta: XorName, owner: PublicKey) -> Self {
+        Self { meta, owner }
     }
 
-    /// Return the identifier of the register.
+    /// Return the network name of the register.
     /// This is used to locate the register on the network.
-    pub fn id(&self) -> XorName {
+    pub fn name(&self) -> XorName {
```

Comment:
I think the previous function name of `id` is more accurate, or even change it to `address`

Reply:
Agreed, not a fan of `name` either, but all other NetworkAddress types use `name` for the network xor name:
```rust
            NetworkAddress::ChunkAddress(chunk_address) => chunk_address.name().0.to_vec(),
            NetworkAddress::DbcAddress(dbc_address) => dbc_address.name().0.to_vec(),
            NetworkAddress::RegisterAddress(register_address) => register_address.name().0.to_vec(),
```
I used name for registers as well so it's the same as the others. Although I wouldn't mind renaming the other as well. 
Should we rename them all to `address()` or even `xorname()`?

Reply:
Renamed them all to `xorname()` so we are explicit.